### DISCUSSION TITLE
Filters not working in DRF after upgrade from 21.1 to 22.1

### DISCUSSION THREAD

**Discussion by empusas** [carltongibson/django-filter#1520] [General]
I had a strange problem that the filters in my project where working just fine on my dev system, but not in production. 
I took me a while to realize that I had version 21.1 installed on my dev and 22.1 in Prod.
After reversing the production system back to 21.2 everything worked there as well.

The problem was that the filter was not working et all and the API always gave back all objects. Also in the browsable API the filter button was missing. But there was no message complaining about anything.

I tried to lookup the changes from 21.1 to 22.1, but it seem the latest version in this repository is 21.1, while pipy shows 22.1 and 22.1 was installed by pip.

**Comment by carltongibson** [comment]
See the [change notes](https://github.com/carltongibson/django-filter/blob/main/CHANGES.rst). 

Various long deprecated fallbacks from the update to v2 were removed. See the [migration guide in the docs](https://django-filter.readthedocs.io/en/main/guide/migration.html) to update.

**Comment by empusas** [comment]
Hi,

thank you for the quick reply. But I am a bit confused. The changelog for 22.1 says that only warnings were removed. So I expected not any impacting changes.
I followed the guide for DRF in the docs https://django-filter.readthedocs.io/en/main/guide/rest_framework.html  and I cannot see anything that changed there.

I am using Django 4.0.6, djangorestframework 3.13.1 and python 3.9.9

This is my code. Works fine with 21.1, but not with 22.1:

Filters.py
 
```
from django_filters import rest_framework as filters
from rest_framework import generics
 
class ACEFilter(filters.FilterSet):
    tenant = filters.CharFilter(field_name="acl__device__tenant__name")
    tenant_id = filters.NumberFilter(field_name="acl__device__tenant__id")
    device = filters.CharFilter(field_name="acl__device__name")
    device_id = filters.NumberFilter(field_name="acl__device__id")
    ruleset = filters.CharFilter(field_name="acl__name")
    ruleset_id = filters.NumberFilter(field_name="acl__id")
 
    class Meta:
        model = ACE
        fields = [
            "id",
            "tenant",
            "tenant_id",
            "device",
            "device_id",
            "ruleset",
            "ruleset_id",
        ]
 
```
 
Views.py
 ```

from django_filters import rest_framework as filters
from rest_framework import generics
from fw_api.filters import ACEFilter
 
class NewACEViewset(viewsets.ReadOnlyModelViewSet):
    """Read only Viewset for new ACE API format"""
 
    authentication_classes = (
        TokenAuthentication,
        SessionAuthentication,
        BasicAuthentication,
    )
    permission_classes = [IsAuthenticated]
    queryset = ACE.objects.all()
    serializer_class = serializers.New_ACESerializer
    filter_class = ACEFilter
    filter_backends = (filters.DjangoFilterBackend,)
    pagination_class = LargeResultsSetPagination

```

**Reply by carltongibson** [reply]
Try clicking the link to the migration guide I gave you and reading it. 

https://django-filter.readthedocs.io/en/main/guide/migration.html#view-attributes-renamed-867

The code here has been emitting deprecation warnings with a fallback since 2017. Plenty of time to update your code. 

Those fallbacks have now been removed. You'll need to update your code.

**Comment by empusas** [comment]
It is not that I am not willing to change my code. I just don´t get what I have to change.
I  don´t see the difference between my code and the example provided in the documenattion.

```
from rest_framework import generics
from django_filters import rest_framework as filters
from myapp import Product


class ProductFilter(filters.FilterSet):
    min_price = filters.NumberFilter(field_name="price", lookup_expr='gte')
    max_price = filters.NumberFilter(field_name="price", lookup_expr='lte')

    class Meta:
        model = Product
        fields = ['category', 'in_stock']


class ProductList(generics.ListAPIView):
    queryset = Product.objects.all()
    serializer_class = ProductSerializer
    filter_backends = (filters.DjangoFilterBackend,)
    filterset_class = ProductFilter
```

**Reply by carltongibson** [reply]
Here's the diff between the two tags. There's not much changed, so there's not much it could be. 

https://github.com/carltongibson/django-filter/compare/21.1...22.1

The most likely commit is https://github.com/carltongibson/django-filter/commit/3dfed5d6755f6e5e25e967e371eb61774eeed015

**Comment by empusas** [comment]
I did not see it all the time.

I just had to replace "filter_class" with "filter**set**_class", then everything worked again.

Thanks for the support.

**Reply by carltongibson** [reply]
Super. Glad you figured it out. Well done!
